mkosi/gentoo.py

```diff
@@ -0,0 +1,508 @@
+                print(f"{c} = {self.portage_cfg[c]}")
+
+        jobs = multiprocessing.cpu_count()
+        self.emerge_default_opts = [
+            "--buildpkg=y",
+            "--usepkg=y",
+            "--keep-going=y",
+            f"--jobs={jobs}",
+            f"--load-average={jobs-1}",
+            "--nospinner",
+        ]
+        if "build-script" in ARG_DEBUG:
+            self.emerge_default_opts += {"--verbose"}
+            self.emerge_default_opts += {"--quiet=n"}
+            self.emerge_default_opts += {"--quiet-fail=n"}
+        else:
+            self.emerge_default_opts += {"--quiet-build"}
+            self.emerge_default_opts += {"--quiet"}
+
+        self.EMERGE_UPDATE_OPTS = {
+            "--update",
+            "--tree",
+            "--changed-use",
+            "--newuse",
+            "--deep",
+            "--with-bdeps=y",
+            "--complete-graph-if-new-use=y",
+        }
+
+        # 'systemd' is hard dependancy?
+        # 'git' for sync-type=git
+        # 'symlink' for kernel
+        self.portage_use_flags = {
+            "systemd",
+            "initramfs",
+            "git",
+            "symlink",
+            "sdl",
+            "-filecaps",
+            "-savedconfig",
+            "-split-bin",
+            "-split-sbin",
+            "-split-usr",
+        }
+
+        # -user* are required for access to USER_CONFIG_PATH
+        # -pid-sandbox is required for cross compile scenarios
+        self.portage_features = {
+            "-userfetch",
+            "-userpriv",
+            "-usersync",
+            "-usersandbox",
+            "-sandbox",
+            "-pid-sandbox",
+            "-network-sandbox",
+            "parallel-install",
+            "buildpkg",
+            "binpkg-multi-instance",
+            "-binpkg-docompress",
+            "getbinpkg",
+            "-candy",
+        }
+        # TODO: portage_features.add("ccache")
+        os.environ["FEATURES"] = " ".join(self.portage_features)
+        os.environ["BOOTSTRAP_USE"] = " ".join(self.portage_use_flags)
+        os.environ["USE"] = " ".join(self.portage_use_flags)
+        os.environ["EGIT_CLONE_TYPE"] = "shallow"
+
+        self.ARCHITECTURES = {
+            "x86_64": ("amd64", "arch/x86/boot/bzImage"),
+            "aarch64": ("arm64", "arch/arm64/boot/Image.gz"),
+            "armv7l": ("arm", "arch/arm/boot/zImage"),
+        }
+        global kimg_path
+        self.arch, kimg_path = self.ARCHITECTURES[args.architecture or "x86_64"]
+
+        # GENTOO_UPSTREAM : we only support systemd profiles! and only the no-multilib flaivour , for now;
+        # GENTOO_UPSTREAM : wait for fix upstream: https://bugs.gentoo.org/792081
+        # GENTOO_TODO     : add args.profile switch for this? (multilib vs. nomultilib)
+        self.arch_profile = f"profiles/default/linux/{self.arch}/{args.release}/no-multilib/systemd"
+
+        self.pkgs_sys = {"@world"}
+
+        self.pkgs_fs = {"sys-fs/dosfstools"}
```

Comment:
`[]` here, and `+= [ … ]` below.

Reply:
done, let me know if i missed something.